aws_lambda_powertools/utilities/parameters/appconf.py

```diff
@@ -0,0 +1,185 @@
+"""
+AWS App Config configuration retrieval and caching utility
+"""
+
+
+import os
+from typing import Dict, Optional, Union
+from uuid import uuid4
+
+import boto3
+from botocore.config import Config
+
+from .base import DEFAULT_PROVIDERS, BaseProvider
+
+CLIENT_ID = str(uuid4())
+
+
+class AppConfigProvider(BaseProvider):
+    """
+    AWS App Config Provider
+
+    Parameters
+    ----------
+    config: botocore.config.Config, optional
+        Botocore configuration to pass during client initialization
+
+    Example
+    -------
+    **Retrieves a parameter value from App Config**
+
+        >>> from aws_lambda_powertools.utilities.parameters import AppConfigProvider
+        >>> appconfig_provider = AppConfigProvider()
+        >>>
+        >>> value: Dict[str, str] = appconfig_provider.get("my-configuration")
+        >>>
+        >>> print(value)
+        My configuration values
+
+    **Retrieves a parameter value from Systems Manager Parameter Store in another AWS region**
+
+        >>> from botocore.config import Config
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>>
+        >>> config = Config(region_name="us-west-1")
+        >>> ssm_provider = SSMProvider(config=config)
+        >>>
+        >>> value = ssm_provider.get("/my/parameter")
+        >>>
+        >>> print(value)
+        My parameter value
+
+    **Retrieves multiple parameter values from Systems Manager Parameter Store using a path prefix**
+
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>> ssm_provider = SSMProvider()
+        >>>
+        >>> values = ssm_provider.get_multiple("/my/path/prefix")
+        >>>
+        >>> for key, value in values.items():
+        ...     print(key, value)
+        /my/path/prefix/a   Parameter value a
+        /my/path/prefix/b   Parameter value b
+        /my/path/prefix/c   Parameter value c
+
+    **Retrieves multiple parameter values from Systems Manager Parameter Store passing options to the SDK call**
+
+        >>> from aws_lambda_powertools.utilities.parameters import SSMProvider
+        >>> ssm_provider = SSMProvider()
+        >>>
+        >>> values = ssm_provider.get_multiple("/my/path/prefix", MaxResults=10)
+        >>>
+        >>> for key, value in values.items():
+        ...     print(key, value)
+        /my/path/prefix/a   Parameter value a
+        /my/path/prefix/b   Parameter value b
+        /my/path/prefix/c   Parameter value c
```

Comment:
These examples use the SSMProvider instead of AppConfigProvider.

Reply:
yep, i know, i wrote that this is a draft pr. need to sort out the comments, docs and tests. Just wanted to know if the general direction/implementation is fine.